src/common/debug.c

```diff
@@ -163,10 +165,26 @@ daos_debug_init(char *logfile)
 	else if (logfile == NULL)
 		logfile = DAOS_LOG_DEFAULT;
 
+	log_file_pid_append = getenv(D_LOG_FILE_APPEND_PID_ENV);
```

Comment:
Am not sure why we are doing the processing here given that cart will append pid already if this env is set; as such you will end up with log files that have pid appended twice it seems

Reply:
This is due to the reference counting in d_log_refcount by d_log_init_adv() this has also to be done in daos_debug_init() which is called before d_log_init() by server_init(). May be I should also remove this possibly redundant call of d_log_init() by server_init() ?

Reply:
Well in fact, I was wrong with the fact that d_log_init() is directly called by server_init(), it is crt_init_opt(), so nothing to be removed.
@frostedcmos , do you agree about what I described before, concerning the fact that twice append of pid would not occur due to the reference counting with d_log_refcount by d_log_init_adv() ??

Reply:
Reading through code it appears we should probably be instead processing this environmental flag inside of d_log_open() call, instead of processing it both here and in d_log_init(). 

This way if some other place later calls d_log_init_adv() it will take this environmental flag into account on its own instead of having to handle it outside

Reply:
> we should probably be instead processing this environmental flag inside of d_log_open() call

Ok, will o in next patch-set